### PR TITLE
Support Yandex Managed Service for Kubernetes

### PR DESCRIPTION
**Pull request by mirtov-alexey**
Good afternoon!
I have prepared a pull request to follow up on discussion # 1065 https://github.com/aquasecurity/kube-bench/discussions/1065. I took the CIS-1.20 profile as a basis and prepared a profile for Yandex Managed Service for Kubernetes. To do this, i only copy the files and add the paths to the config file to your config.
I have attached the verification log and the source files themselves from our k8s node to the folder "files for discussion". I can also provide instructions for deploying a cluster in our cloud, or arrange an online session if the logs are not enough.

Installed from here curl -L https://github.com/aquasecurity/kube-bench/releases/download/v0.6.5/kube-bench_0.6.5_linux_amd64.tar.gz -o kube-bench_0.6.5_linux_amd64.tar.gz

Run like this: ./kube-bench run --config-dir ./cfg --config ./cfg/config.yaml --targets node --benchmark yc-1.0

I faced the following problems:
• Checks 4.1.3, 4.1.4, 4.1.7, 4.1.8, 4.2.4, 4.2.8 (MANUAL) shows the status “PASS” although according to your documentation https://github.com/aquasecurity/kube-bench /blob/main/docs/flags-and-commands.md MANUAL checks should always show the status "WARN". How can you comment on this?
• 4.2.5 "PASS" status expects streamingConnectionIdleTimeout not equal to 0, but in fact the "streamingConnectionIdleTimeout: 0s" parameter is specified in the /home/kubernetes/kubelet-config.yaml file, which contradicts the condition and should be "FAIL"
• 4.2.7 the same situation - shows "PASS", but in fact the parameter in our config "makeIPTablesUtilChains" is not presented
• 4.2.12 the same situation - shows "PASS", but in fact the parameter in our config "RotateKubeletServerCertificate" is not presented
• Also, for comparison, I checked the triggers on my test cluster in aws cloud with the usual CIS profile. Regarding point: 4.2.5 also shows the "PASS" status, but the streamingConnectionIdleTimeout parameter is completely absent in the /etc/kubernetes/kubelet/kubelet-config.json file. According to clause 4.2.7, the same is true.
• Clause: 4.2.6, which we have FAIL we will correct on our side, like others

Please look and check, after we solve these problems - I will attach confirmation from Yandex Clouds and the final config and we can move on. Thanks a lot.

Additionally, the command output is:

root@cl14kmk32gc3pstvbaae-ucuz:~# ps -ef | grep kubelet
root         984       1  2 Dec01 ?        04:50:19 /home/kubernetes/bin/kubelet --bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubeconfig.conf --cert-dir=/var/lib/kubelet/pki/ --cloud-provider=external --cni-bin-dir=/home/kubernetes/cni/bin --config=/home/kubernetes/kubelet-config.yaml --experimental-kernel-memcg-notification=true --feature-gates=CSINodeInfo=true,CSIDriverRegistry=true --kubeconfig=/etc/kubernetes/kubelet-kubeconfig.conf --network-plugin=kubenet --node-status-max-images=25 --non-masquerade-cidr=0.0.0.0/0 --pod-infra-container-image=cr.yandex/crpsjg1coh47p81vh2lc/pause:3.1 --resolv-conf=/run/systemd/resolve/resolv.conf --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_128_GCM_SHA256 --v=2 --node-labels=node.kubernetes.io/kube-proxy-ds-ready=true,node.kubernetes.io/masq-agent-ds-ready=true,node.kubernetes.io/node-problem-detector-ds-ready=true,yandex.cloud/node-group-id=cat1gdai8kvqiimg7tnh,yandex.cloud/pci-topology=k8s,yandex.cloud/preemptible=false
root        1729    1708  0 Dec01 ?        00:00:02 /csi-node-driver-registrar --csi-address=/csi/csi.sock --kubelet-registration-path=/var/lib/kubelet/plugins/disk-csi-driver.mks.ycloud.io/csi.sock --v=5
root        1881    1860  0 Dec01 ?        00:00:02 /csi-node-driver-registrar --csi-address=/csi/csi.sock --kubelet-registration-path=/var/lib/kubelet/plugins/io.ycloud.mks.disk-csi-driver/csi.sock --v=5
root        2675    2654  0 Dec01 ?        00:07:16 /metrics-server --metric-resolution=30s --kubelet-insecure-tls --kubelet-preferred-address-types=InternalIP
root     2988944 2988738  0 14:32 pts/0    00:00:00 grep --color=auto kubelet

### CODE REVIEW

**Comment by yoavrotems** [comment]
Hey @mirtov-alexey ,
Thanks for contributing it is always nice to support a new platform 🚀 
> To do this, i only copy the files and add the paths to the config file to your config.

If I understood correctly you didn't changed cis yaml files but just added new paths? If so then you don't need a new spec, It's enough to add the paths to config.yaml and it will be auto-selected if its exist. (Config.yaml works top-bottom so if there will be another match with config files the first to appear is the one kube-bench will assume is in use).
Just like we did with BottlerocketOS #889 

> I can also provide instructions for deploying a cluster in our cloud

More data and examples are always good 😸 

> • Checks 4.1.3, 4.1.4, 4.1.7, 4.1.8, 4.2.4, 4.2.8 (MANUAL) shows the status “PASS” although according to your documentation https://github.com/aquasecurity/kube-bench /blob/main/docs/flags-and-commands.md MANUAL checks should always show the status "WARN". How can you comment on this? 

You are basically right! but.. in the latest CIS versions there is a change in terminology, tests used to be Scored/Not Scored. But they changed it to be Automated/Manual which was almost direct change Scored -> Automated, Not Scored -> Manual with a few exceptional. 
So here there are tests that CIS calls XXX(Manual) but we do have automation for them. 
What is defined as manual tests that will produce WARN status are the tests with `type: manual` in the yaml. 

> • 4.2.5 "PASS" status expects streamingConnectionIdleTimeout not equal to 0, but in fact the "streamingConnectionIdleTimeout: 0s" parameter is specified in the /home/kubernetes/kubelet-config.yaml file, which contradicts the condition and should be "FAIL"

Could be a false positive (I Guess the 's' could interrupt) I would recommend to open an issue in separate about this and let check it 👓 

> • 4.2.7 the same situation - shows "PASS", but in fact the parameter in our config "makeIPTablesUtilChains" is not presented
• 4.2.12 the same situation - shows "PASS", but in fact the parameter in our config "RotateKubeletServerCertificate" is not presented

Here if I'm not mistaken if not present default value is good, there is more information in the CIS. 
Sometime we check for the existent of a parameter and if so the value, but if its not config then we know the default kubernetes value is ok. There could be mistakes there but we will need to open a ticket about it in CIS official site, because we follow their guidelines.

**Comment by mirtov-alexey** [comment]
@yoavrotems Hello!
Thank you very much for your response!
According to your answer, I really only left the changes in the config.yaml file and deleted everything unnecessary.
I am attaching the results of how I was tested with the current config and what the results were after adding new paths.

Before:
== Summary node ==
18 checks PASS
2 checks FAIL
3 checks WARN
0 checks INFO

After:
== Summary node ==
20 checks PASS
1 checks FAIL
2 checks WARN
0 checks INFO

According to the results, we have only one test left that we do not pass - this is 4.2.6, but we are already working on ensuring that our nodes pass it.

Here I am enclosing instructions on how to set up a demo cluster in our cloud to check it - https://cloud.yandex.com/en/docs/managed-kubernetes/quickstart

but to speed up the time, I can also drop the credentials from the test cluster in a personal message so that you can check it yourself, if necessary

As a result, we could please our clients and the community that now kube-bench also supports the Yandex cloud for verifying nodes

**Comment by mirtov-alexey** [comment]
@yoavrotems Hello!
respectfully remind) please specify when you will have time to look at pr?

**Comment by yoavrotems** [comment]
Hey @mirtov-alexey I'm currently on some time off for the holidays, I will get to it next week, thanks for your patience 😃 
I will get to it next week

**Comment by mozillazg** [comment]
LGTM

File: cfg/config.yaml [hunk 1]
```diff
@@ -312,4 +314,4 @@ target_mapping:
     - "node"
     - "controlplane"
     - "policies"
-    - "etcd"
+    - "etcd"
```
Comment (mozillazg) [inline cfg/config.yaml:317]: There is a yaml-lint error:
`config.yaml:317:13: [error] no new line character at the end of file (new-line-at-end-of-file)`

https://github.com/aquasecurity/kube-bench/runs/4663081767?check_suite_focus=true

Comment (mirtov-alexey) [reply cfg/config.yaml]: @mozillazg 
Hello!
i fixed this issue and pushed the new file
please let me try again)
maybe we could do it before new year like a christmas gift)

Comment (mozillazg) [reply cfg/config.yaml]: @mirtov-alexey hi, there are some needless white space:



https://github.com/aquasecurity/kube-bench/runs/4664605716?check_suite_focus=true

>  Error: onfig.yaml:318:5: [error] no new line character at the end of file (new-line-at-end-of-file)
> Error: onfig.yaml:318:1: [error] trailing spaces (trailing-spaces)
> Error: Process completed with exit code 1.

Comment (mirtov-alexey) [reply cfg/config.yaml]: @mozillazg fixed it)sorry its my first pr in public repos

**Review by mozillazg — APPROVED** [review]
LGTM!

**Comment by mozillazg** [comment]
@itaysk ping~
